test/algorithms/classifiers/test_vqc.py

```diff
@@ -35,367 +114,172 @@ class TestVQC(QiskitMachineLearningTestCase):
 
     def setUp(self):
         super().setUp()
-
+        algorithm_globals.random_seed = RANDOM_SEED
```

Comment:
Ah, yes, I see. So, the reason I defined them outside setup is so I could use them directly with `ddt`. I'd actually prefer to refactor the tests to avoid `ddt` and just use native `unittest` methods. But, `ddt` is used elsewhere in the package with globals, so I wanted to keep in line with those (though they may be mutable in these cases). Do you know off the top of your head if I can use variables from setup directly with ddt (without using string flags). Or else are there any objections to refactoring away from ddt?

Reply:
You could define functions, say on the test class, and have these be called by ddt to create the instances on demand. You could also do it in setup like before but say place the instances into dictionary and use places a key in ddt to lookup the instance you want - these could be string keys. This would then be a simple key lookup rather than the if tests, based on the strings, that were there before.

We use ddt where we have the same test in effect but different values for it - the values can be inputs as well as expected results given the different inputs if the expected result varies. Given a want to exercise some code the same way over a set of values is why we use it to decorate the tests - it then under the covers essentially expands out everything to create all the needed tests each one run separately. Yes you can use for loops along with sub tests to approximate this so the test loops over all conditions and each being a sub-test will still run even if one of them fails. We used to use a different package to do this in the past but went over to ddt which is what Terra decided to use.

Reply:
So, I opted to use `subTest()`s instead of `ddt` functions. Mostly because I think it's slightly cleaner and easier to read in this case. I've also moved all of the setups into `setUp`. Hopefully, this means the subtests are set up correctly now.

Reply:
Please see [this](https://github.com/Qiskit/qiskit-machine-learning/pull/374#discussion_r875142649) comment for further discussion.